### DISCUSSION TITLE
brew config reporting strange Xcode CLT version

### DISCUSSION THREAD

**Discussion by DanielKehoe** [Homebrew/discussions#1784] [Getting started]
What is the correct way to determine the Xcode Command Line Tools version? 

When I use `xcode-select --version` I get `version 2384` on macOS 11.3 (Big Sur). But when I run `brew config` it reports CLT version 12.5.0.22.9. Very different strings. Why does `brew config` report something different from `xcode-select --version`?

**Comment by carlocab** [comment]
`xcode-select --version` is not the version of Xcode you have installed. `xcode-select --version` is the version of `xcode-select`.

**Reply by DanielKehoe** [reply]
Thank you for the clarification. How can I see the version of Xcode from in the terminal without using `brew config`?

**Reply by carlocab** [reply]
You could try `brew ruby -e 'puts MacOS::Xcode.version'`. Or, check `clang --version` and check the output against https://en.wikipedia.org/wiki/Xcode#Xcode_11.x_-_13.x_(since_SwiftUI_framework)_2.

**Reply by DanielKehoe** [reply]
Is there a way to get the same output `12.5.0.22.9` as reported by `brew config` without using brew? Can `12.5.0.22.9` be obtained directly from the command line without brew?

**Reply by SMillerDev** [reply]
https://dive.medium.com/how-to-get-the-current-selected-xcode-and-list-installed-e16278c8c65a this was the first google result when I searched for "get xcode version"

**Reply by SMillerDev** [reply]
And in my own scripts I use https://github.com/M2mobi/jenkins-node-labeler/blob/master/labeler.py#L22-L23

**Reply by DanielKehoe** [reply]
Ooops I think I introduced a misunderstanding. But thank you for the effort!

I've got Xcode Command Line Tools installed, not the full Xcode package (I'm not developing macOS software). So I'm specifically trying to find if there's a way to determine the version of the installed Xcode Command Line Tools, not Xcode. As a diagnostic tool, I'm trying to find if the command line can be used to determine the installed Xcode Command Line Tools version (not the XCode version). Trying to see it can be found without running brew. And wondering how brew gets the version number.

**Reply by DanielKehoe** [reply]
Just to explain why I'm asking, I'm writing a guide to install Homebrew, and for people who already have Xcode Command Line Tools installed, I want to provide a diagnostic so they can know if they need to upgrade Xcode Command Line Tools before installing Homebrew.

**Reply by SMillerDev** [reply]
Brew will tell them, or install it for them if needed. You're making the install process a lot more complicated than it needs to be.

**Reply by DanielKehoe** [reply]
Yes, I see that now. Just saw a lot of people searching for how to determine the Xcode Command Line Tools version and wondered how brew config does it and if there's a command line way to do it. Surprisingly difficult to find this answer. But I appreciate your efforts and suggestions.

**Reply by carlocab** [reply]
`brew` parses this from the output of `clang --version`. See https://github.com/Homebrew/brew/blob/35b15af4005ff0140a73a64e48abb4cbc85cc270/Library/Homebrew/os/mac/xcode.rb.

There's really no need to check the CLT version before installing `brew`, though. Just install `brew` and then do `brew doctor` to check if there's anything you need to fix.
